### PR TITLE
OCPBUGS-13649: add tests for ClusterResourceQuota: count should work for all resources

### PR DESCRIPTION
**Pull request by atiratree**

### CODE REVIEW

File: test/extended/quota/clusterquota.go [hunk 1]
```diff
@@ -18,21 +18,27 @@ import (
...
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 var _ = g.Describe("[sig-api-machinery][Feature:ClusterResourceQuota]", func() {
 	defer g.GinkgoRecover()
 	oc := exutil.NewCLI("crq")
 
 	g.Describe("Cluster resource quota", func() {
-		g.It(fmt.Sprintf("should control resource limits across namespaces [apigroup:quota.openshift.io][apigroup:image.openshift.io]"), func() {
+		g.It(fmt.Sprintf("should control resource limits across namespaces [apigroup:quota.openshift.io][apigroup:image.openshift.io][apigroup:monitoring.coreos.com][apigroup:template.openshift.io]"), func() {
```
Comment (atiratree) [inline test/extended/quota/clusterquota.go:34]: @ingvagabund do you see a problem with increasing the scope of resouces this test needs? Mainly the monitoring CRD I guess. Should I abstract this and split this into two tests?

File: test/extended/quota/clusterquota.go [hunk 2]
```diff
@@ -189,6 +199,74 @@ var _ = g.Describe("[sig-api-machinery][Feature:ClusterResourceQuota]", func() {
 
 				t.Fatalf("unexpected error: %v", err)
 			}
+
+			// test templates are counted correctly
+			template := &templatev1.Template{}
```
Comment (atiratree) [inline test/extended/quota/clusterquota.go:204]: this test confirms the already merged https://github.com/openshift/cluster-policy-controller/pull/100

File: test/extended/quota/clusterquota.go [hunk 3]
```diff
@@ -189,6 +199,74 @@ var _ = g.Describe("[sig-api-machinery][Feature:ClusterResourceQuota]", func() {
...
+				list2, err := clusterAdminTemplateClient.TemplateV1().Templates("").List(context.Background(), metav1.ListOptions{})
+				if err == nil {
+					t.Errorf("Templates is %#v", list2)
+				}
+
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			// test that CRD resources are counted correctly
+			serviceMonitorGVR := schema.GroupVersionResource{Group: "monitoring.coreos.com", Version: "v1", Resource: "servicemonitors"}
```
Comment (atiratree) [inline test/extended/quota/clusterquota.go:238]: this should start passing once https://github.com/openshift/cluster-policy-controller/pull/115 merges

File: test/extended/quota/clusterquota.go (hunk 1 above)
Comment (ingvagabund) [reply test/extended/quota/clusterquota.go:34]: This is acceptable. The test can be refined later once needed.

**Comment by atiratree** [comment]
[[sig-api-machinery][Feature:ClusterResourceQuota] Cluster resource quota should control resource limits across namespaces [apigroup:quota.openshift.io][apigroup:image.openshift.io][apigroup:monitoring.coreos.com][apigroup:template.openshift.io] [Suite:openshift/conformance/parallel]](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27934/pull-ci-openshift-origin-master-e2e-gcp-ovn/1661614747759415296) passing

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**